apps/atomy-q/API/app/Adapters/QuotationIntelligence/DeterministicContentProcessor.php

```
@@ -74,12 +74,42 @@ public function analyze(string $storagePath): object
-        $prefix = rtrim(Storage::disk('local')->path(''), DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
-        if (str_starts_with($storagePath, $prefix)) {
-            return substr($storagePath, strlen($prefix));
+        $normalizedPath = str_replace('\\', '/', $storagePath);
+
+        foreach ($this->storageRootPrefixes() as $prefix) {
+            if (str_starts_with($normalizedPath, $prefix)) {
+                return ltrim(substr($normalizedPath, strlen($prefix)), '/');
+            }
+        }
+
+        $quoteSubmissionSegment = '/quote-submissions/';
```

PHPMD (2.15.0)

[warning] Avoid excessively long variable names like $quoteSubmissionSegment. Keep variable name length under 20. (undefined)

(LongVariable)